runtime/interpreter/encode.go

```diff
@@ -1006,6 +1016,11 @@ func (e *Encoder) encodeCompositeValue(
 		return err
 	}
 
+	// Pre-allocate and reuse valuePath.
+	valuePath := append(path, "")
```

Comment:
Same as above: `append` mutates the argument, so this would effect the calling code. The source argument should be a copy:

```suggestion
	valuePath := append(path[:], "")
```

This is still better than the original, which copied for each element

Reply:
Resolving as per discussion below